beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3702,8 +3760,58 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         match availability {
             Availability::Available(block) => {
                 publish_fn()?;
-                // Block is fully available, import into fork choice
-                self.import_available_block(block).await
+                let block_root = block.block_root();
+                match self.block_status_table.transition(
+                    &block_root,
+                    BlockStatus::Pending,
+                    BlockStatus::Importing,
+                ) {
+                    Ok(_) => {
+                        let result = self.import_available_block(block).await;
+                        if let Ok(AvailabilityProcessingStatus::Imported(_)) = result.as_ref() {
+                            self.block_status_table
+                                .transition(
+                                    &block_root,
+                                    BlockStatus::Importing,
+                                    BlockStatus::Imported,
+                                )
+                                .map_err(|e| {
+                                    BlockError::InternalError(format!(
+                                        "BlockStatusTable failed on transition: {:?}",
+                                        e
+                                    ))
+                                })?;
+                        } else {
+                            // something went wrong on import, mutate the status table back to pending
+                            // TODO: determine when we would mutate this to Invalid!
+                            self.block_status_table
+                                .transition(
+                                    &block_root,
+                                    BlockStatus::Importing,
+                                    BlockStatus::Pending,
+                                )
+                                .map_err(|e| {
+                                    BlockError::InternalError(format!(
+                                        "BlockStatusTable failed on transition: {:?}",
+                                        e
+                                    ))
+                                })?;
+                        }
+
+                        result
+                    }
+                    Err(e) => match e {
+                        // someone beat us to it - the block is already being imported
+                        BlockStatusTableError::InvalidStatusTransition { .. } => {
+                            Err(BlockError::DuplicateFullyImported(block_root))
+                        }
```

Comment:
Not true, this error will trigger if the current status is != Pending. If that status happens to be `Invalid` you will return `DuplicateFullyImported` which is incorrect